source/MaterialXTest/Runtime.cpp

```diff
@@ -1675,4 +1675,37 @@ TEST_CASE("Runtime: libraries", "[runtime]")
     REQUIRE(api->getImplementationSearchPath().find("stdlib_genglsl_unit_impl.mtlx").exists());    
 }
 
+TEST_CASE("Runtime: units", "[runtime]")
+{
+    mx::RtScopedApiHandle api;
+
+    // Load in all libraries required for materials
+    mx::FileSearchPath searchPath(mx::FilePath::getCurrentPath() / mx::FilePath("libraries"));
+    api->setSearchPath(searchPath);
+    api->loadLibrary(STDLIB);
+    api->loadLibrary(PBRLIB);
+    api->loadLibrary(BXDFLIB);
+
+    // Read in test document with units
+    mx::FileSearchPath testSearchPath(mx::FilePath::getCurrentPath() /
+        "resources" /
+        "Materials" /
+        "TestSuite" /
+        "stdlib" /
+        "units");
+    mx::RtStagePtr stage = api->createStage(mx::RtToken("stage"));
+    mx::RtFileIo fileIo(stage);
+    mx::StringVec tests{ "distance_units.mtlx",
```

Comment:
"Unit" related units now load properly without issue.

Reply:
minor: Should you have a unit test for writing units information as well?

Reply:
I was only checking nodedefs. I've update the code to load/save node unit information properly now. 
I noticed colorspace attribute on nodes is not loaded/saved properly but not addressing here.